zerver/models/recipients.py

```diff
@@ -191,3 +191,21 @@ def get_or_create_direct_message_group(id_list: list[int]) -> DirectMessageGroup
             ]
             Subscription.objects.bulk_create(subs_to_create)
         return direct_message_group
+
+
+def get_direct_message_group(id_list: list[int]) -> DirectMessageGroup | None:
+    """
+    Takes a list of user IDs and returns the DirectMessageGroup
+    object for the group consisting of these users if exists. If
+    the DirectMessageGroup object does not yet exist, it will
+    return None.
+    """
+
+    direct_message_group_hash = get_direct_message_group_hash(id_list)
+    direct_message_groups = DirectMessageGroup.objects.filter(
+        huddle_hash=direct_message_group_hash,
+        group_size=len(id_list),
+    )
+    if not direct_message_groups.exists():
+        return None
+    return direct_message_groups.first()
```

Comment:
I think you might be able to optimize this by just querying `.first()`? That might just already return `None` if nothing matches; but if not, we can catch the exception and return; in either case, we save a query when it does exist.

Reply:
Better would be to do `.get`, and catch `DirectMessageGroup.DoesNotExist` to return `None` -- then we throw an exception if there's more than one (which would be invalid/corrupted database state).

Reply:
Great point. Updated.